src/activityLog/Activity.ts

```diff
@@ -0,0 +1,58 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtTreeItem, IParsedError, parseError } from "@microsoft/vscode-azext-utils";
+import { TreeItemCollapsibleState } from "vscode";
+
+export interface ActivityTreeItemOptions {
+    label: string;
+    contextValuePostfix?: string;
+    collapsibleState?: TreeItemCollapsibleState;
+    children?: (parent: AzExtParentTreeItem) => AzExtTreeItem[];
+}
+
+interface ActivityType {
```

Comment:
Maybe just rename this `Activity`? (Renaming will affect some suggestions below)
```suggestion
interface Activity {
```
And export it so consumers of the API can use it instead of `ActivityBase`?